source/backend/dal/Repositories/DispositionFileRepository.cs

```diff
@@ -344,12 +346,46 @@ public PimsDispositionSale AddDispositionFileSale(PimsDispositionSale dispositio
         public PimsDispositionSale UpdateDispositionFileSale(PimsDispositionSale dispositionSale)
         {
             var existingSale = Context.PimsDispositionSales
+                .Include(x => x.DspPurchAgent)
+                .Include(x => x.DspPurchSolicitor)
                 .FirstOrDefault(x => x.DispositionSaleId.Equals(dispositionSale.DispositionSaleId)) ?? throw new KeyNotFoundException();
 
+            if (existingSale.DspPurchAgent != null && dispositionSale.DspPurchAgent == null)
+            {
+                Context.Remove(existingSale.DspPurchAgent);
+                dispositionSale.DspPurchAgentId = null;
+            }
```

Comment:
Test coverage on these conditionals is a bit spotty, can you add some tests for this?

Reply:
updated